src/webots/gui/WbConsole.cpp

```diff
@@ -272,29 +272,35 @@ QString WbConsole::htmlSpan(const QString &s, WbLog::Level level) const {
   if (s.isEmpty() || s == "\n")
     return "";
 
-  QString color;
   bool bold;
+  QString foregroundColor;
   if (level == WbLog::ERROR || level == WbLog::FATAL || level == WbLog::STDERR) {
-    color = errorColor();
+    foregroundColor = errorColor();
     bold = true;
   } else if (level == WbLog::WARNING || level == WbLog::DEBUG) {
-    color = errorColor();
+    foregroundColor = errorColor();
     bold = false;
   } else if (level == WbLog::INFO || level == WbLog::STATUS) {
-    color = infoColor();
+    foregroundColor = infoColor();
     bold = false;
   } else {
     assert(level == WbLog::STDOUT);
-    color = mColor;
+    foregroundColor = mForegroundColor;
     bold = mBold;
   }
   QString span("<span");
-  if (!color.isEmpty() || bold) {
+  if (!foregroundColor.isEmpty() || bold || mUnderline) {
     span += " style=\"";
-    if (!color.isEmpty())
-      span += "color:" + color + ";";
+    if (!foregroundColor.isEmpty() && mBackgroundColor.isEmpty())
+      span += "color:" + foregroundColor + ";";
+    else if (!foregroundColor.isEmpty() && !mBackgroundColor.isEmpty()) {
+      span += "color:" + foregroundColor + ";";
+      span += "background-color:" + mBackgroundColor + ";";
+    }
```

Comment:
That part of code looks odd to me. Can't we simplify it like that?
```suggestion
    if (!foregroundColor.isEmpty())
      span += "color:" + foregroundColor + ";";
    if (!mBackgroundColor.isEmpty())
      span += "background-color:" + mBackgroundColor + ";";
```

Reply:
True it is more clear, on it!